pinot-core/src/test/java/org/apache/pinot/util/TestUtils.java

```diff
@@ -140,35 +140,32 @@ public static void ensureDirectoriesExistAndEmpty(@Nonnull File... dirs)
    * @param timeoutMs Timeout in milliseconds
    * @param errorMessage Error message if condition is not met before timed out
    */
-  public static void waitForCondition(@Nonnull Function<Void, Boolean> condition, long checkIntervalMs, long timeoutMs,
-      @Nullable String errorMessage)
-      throws Exception {
+  public static void waitForCondition(Function<Void, Boolean> condition, long checkIntervalMs, long timeoutMs,
```

Comment:
I think it's unnecessary to refactor this method. It's better to put while loop inside the try block. Plus, in your current way you'll need two places to handle `errorMessage != null` cases.

Reply:
You should not put while loop inside the try-catch because the outer catch will catch the exception thrown out from the inner loop. Merged the error message check to one place (in the old code we also have two places to handle `errorMessage != null`).